src/Polly.Core/Simmy/MonkeyStrategy.TResult.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Polly.Simmy.Utils;
+
+namespace Polly.Simmy;
+
+/// <summary>
+/// This base strategy class is used to simplify the implementation of generic (reactive)
+/// strategies by limiting the number of generic types the execute method receives.
+/// </summary>
+/// <typeparam name="T">The type of result this strategy handles.</typeparam>
+/// <remarks>
+/// For strategies that handle all result types the generic parameter must be of type <see cref="object"/>.
+/// </remarks>
+public abstract class MonkeyStrategy<T> : ResilienceStrategy<T>
+{
+    private readonly Func<double> _randomizer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MonkeyStrategy{T}"/> class.
+    /// </summary>
+    /// <param name="options">The chaos strategy options.</param>
+    protected MonkeyStrategy(MonkeyStrategyOptions options)
+    {
+        Guard.NotNull(options);
+        Guard.NotNull(options.Randomizer);
+
+        _randomizer = options.Randomizer;
+        InjectionRateGenerator = options.InjectionRateGenerator is not null ? options.InjectionRateGenerator : (_) => new(options.InjectionRate);
+        EnabledGenerator = options.EnabledGenerator is not null ? options.EnabledGenerator : (_) => new(options.Enabled);
+    }
+
+    /// <summary>
+    /// Gets the injection rate for a given execution, which the value should be between [0, 1].
+    /// </summary>
+    public Func<InjectionRateGeneratorArguments, ValueTask<double>> InjectionRateGenerator { get; }
```

Comment:
Does anyone actually needs these properties? Won't everyone ever use just `ShouldInjectAsync`?

Reply:
Do you think is better to hide them? They're exposed as read-only properties as the old version exposes them

Reply:
The approach we took is to hide stuff we think/know it's not required. It's easier to expose something than to hide/obsolete it later.